runtime/runtime/src/actions.rs

```diff
@@ -687,11 +714,74 @@ pub(crate) fn check_account_existence(
     Ok(())
 }
 
+#[cfg(feature = "protocol_feature_add_account_versions")]
+pub fn recalculate_usage(
+    account: &mut Account,
+    account_id: &AccountId,
+    usage_config: &StorageUsageConfig,
+    state_update: &TrieUpdate,
+) -> Result<(), StorageError> {
+    if let Account::AccountV1(_) = account {
+        // As this is valid account that fits into state we can use unchecked operations here
+        let mut usage: StorageUsage = usage_config.num_bytes_account;
+        if let Some(code) = get_code(state_update, account_id, Some(account.code_hash()))? {
+            usage += code.code.len() as u64;
+        }
+        state_update
```

Comment:
what is the reason for going over all the account storage? Couldn't we simply check the access key part?

Reply:
We do not store storage usage by type anywhere, only total